packages/psss/src/utils/period.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { getCurrentPeriod, convertPeriodStringToDateRange } from '@tupaia/utils';
+import { format, getISODay, subWeeks } from 'date-fns';
+import { DUE_ISO_DAY } from '../constants';
+
+/**
+ *
+ * Convenience utils that build on top of @tupaia/utils
+ * Uses period string (eg. 2020W32)
+ */
+
+export const getDisplayDatesByPeriod = period => {
+  const start = getFormattedStartByPeriod(period, 'LLL d');
+  const end = getFormattedEndByPeriod(period, 'LLL d, yyyy');
+  return `${start} - ${end}`;
+};
+
+export const getFormattedStartByPeriod = (period, f) => {
+  const dates = convertPeriodStringToDateRange(period);
+  return format(new Date(dates[0]), f);
+};
+
+export const getFormattedEndByPeriod = (period, f) => {
+  const dates = convertPeriodStringToDateRange(period);
+  return format(new Date(dates[1]), f);
+};
+
+export const getCurrentIsoWeekNumber = () => {
+  const period = getCurrentPeriod('WEEK');
+  return getWeekNumberByPeriod(period);
+};
+
+export const getWeekNumberByPeriod = period => period.split('W').pop();
+
+export const getDateByPeriod = period => {
+  const date = convertPeriodStringToDateRange(period)[0];
+  return new Date(date);
+};
+
+export const getPeriodByDate = date => format(date, "yyyy'W'II");
```

Comment:
`dateStringToPeriod` from `@tupaia/utils` may also do what you need here, with the advantage of encapsulating the week format

Reply:
I tried this and I will leave it as is because I was required to format the date into a different specific format for the util.  I'd rather hard code the date format than make it a 2 step process and rely on the imported util here. We are committed to the date format for periods in a number of places anyway and I think that is unavoidable.

Reply:
No worries, sounds like a good call!